src/Target.cpp

```diff
@@ -62,6 +62,63 @@ static void cpuid(int info[4], int infoType, int extra) {
 #endif
 #endif
 
+#if defined(__x86_64__) || defined(__i386__) || defined(_MSC_VER)
+
+enum class VendorSignatures {
+    UNKNOWN,
+    GENUINE_INTEL,
+    AUTHENTIC_AMD,
+};
+
+VendorSignatures get_vendor_signature() {
+    int info[4];
+    cpuid(info, 0, 0);
+
+    if (info[0] < 1) {
+        return VendorSignatures::UNKNOWN;
+    }
+
+    // "Genu ineI ntel"
+    if (info[1] == 0x756e6547 && info[3] == 0x49656e69 && info[2] == 0x6c65746e) {
+        return VendorSignatures::GENUINE_INTEL;
+    }
+
+    // "Auth enti cAMD"
+    if (info[1] == 0x68747541 && info[3] == 0x69746e65 && info[2] == 0x444d4163) {
+        return VendorSignatures::AUTHENTIC_AMD;
+    }
+
+    return VendorSignatures::UNKNOWN;
+}
+
+void detect_family_and_model(unsigned info0, unsigned &family,
```

Comment:
`info0` is gathered as an `int` but passed as an `unsigned`. Seems more sensible to just continue to use `int` for all of these here, unless there's a compelling to use `unsigned`.

Reply:
Oh. But we are really interested in (unsigned) bit fields, so sign really isn't a thing.
How about we canonicalize in opposite direction instead?

Reply:
Hm, so i guess it was signed because that is what required for MSVC `__cpuidex()`.
I took liberty of changing to unsigned and adding a wrapper to hopefully keep `__cpuidex()` happy,
but i guess i could go to `int` instead if you insist...

Reply:
It's `int` everywhere else. Let's just use that.

Reply:
Well, okay then.